plugin/src/main/java/io/deephaven/plugin/type/ObjectType.java

```diff
@@ -46,7 +47,85 @@ public interface ObjectType extends Plugin {
      * @param out the output stream
      * @throws IOException if an IO exception occurs
      */
-    void writeTo(Exporter exporter, Object object, OutputStream out) throws IOException;
+    default void writeTo(Exporter exporter, Object object, OutputStream out) throws IOException {
```

Comment:
I think it would be really nice to get rid of this, and instead build an adapter for existing plugins to work in the new way. @devinrsmith maybe we can keep `ObjectType` for the old ones, and add a new name for the new thing. There's a lot of opinionated API here that I would love to disavow.

Reply:
The unary model isn't going away, but could be relegated to an abstract subclass (we're already three deep for some simple java types, so it isn't that much more layering...) that must return a single message and close right away after sending. These are the questions I'm hoping to have answered now, so we can cut old APIs and get something we like better, or decide that we have to keep compat with external plugins that already exist.